src/utils/action-utils.js

```diff
@@ -0,0 +1,24 @@
+/**
+ * Copyright 2018 The Subscribe with Google Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Returns boolean for whether publisher is eligible for gTag
+ * @param {!../runtime/deps.DepsDef} deps
+ * @return {boolean}
+ */
+export function isGtagEligible(deps) {
```

Comment:
I will use this to refactor the changes in https://github.com/subscriptions-project/swg-js/pull/2420 once that is merged.

Reply:
I'm not thrilled about this location. What I'm wondering is if we can just add it as a static function on the `GoogleAnalyticsEventListener` class. I feel like the check should just use something like `(self.gtag || null) == 'function` instead of needing the deps for the window object. @ChrisAntaki do you have a preference

Reply:
I like the idea of defining a static method on the `GoogleAnalyticsEventListener` class, especially if it accepts a `deps` param. We might be able to simplify the conditional logic to `typeof deps.win().gtag === 'function'`

Reply:
Agree, I can create a static method, something else I wanted to do was to refactor this utils in favor of https://github.com/subscriptions-project/swg-js/blob/main/src/utils/types.js#L49

Reply:
Made the change, but one thing I want to double check, can we guarantee that `self.gtag` === `deps.win().gtag` (and similarly for `ga`)?

Reply:
They might differ, but I think that's fine. Sometimes AMP will pass in a `win` object that differences from the current `window` object, right?

Reply:
Futhermore, the changes in https://github.com/subscriptions-project/swg-js/pull/2420 are executing `gtag` from `win`, whereas this check is checking `self`; does this open up the possibility for bad GA logging?

Reply:
Pretty sure we should consistently check the "win" object instead of referencing the current global scope with "self"

Reply:
Could the new static methods both require a `deps` parameter?

Reply:
I think checking `self` (global scope) might lead to issues with AMP

Reply:
+1 to passing `deps` (or `win`) object from callers to avoid the possibility for mis logging.

Reply:
Updated function to accept `deps` parameter.